tests/kernel/virtio_console_long_output.pm

```diff
@@ -0,0 +1,108 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Stress test the virtio serial terminal with long output.
+# Maintainer: cfamullaconrad@suse.com
+
+use Mojo::Base 'opensusebasetest';
+use testapi;
+use utils;
+use Mojo::Util qw(sha1_sum trim);
+use Mojo::File 'path';
+use serial_terminal;
+
+sub create_test_data
+{
+    my $size        = shift // 1024 * 1024;
+    my $line_length = shift // 79;
+    my @a = ('a' .. 'z', 'A' .. 'Z', 0 .. 9);
+
+    my $data = "";
+    for (my $i = 1; $i < $size; $i++) {
+        if ($i > 0 && ($i % $line_length) == 0) {
+            $data .= $/;
+        } else {
+            $data .= $a[($i) % @a];
+        }
+    }
+    $data .= $/;
+    return $data;
+}
+
+sub run {
+    my $self   = shift;
+    my $size   = get_var('VIRTIO_CONSOLE_TEST_FILESIZE') // 200 * 1024;
+    my $repeat = 1000;
+
+    $self->select_serial_terminal;
+
+    # prepare upload directory
+    system('mkdir -p ulogs/') == 0 or die('Failed to create ulogs/ directory');
+
+    my $filename = "original_$size.txt";
+    my $testdata = create_test_data($size);
+    path('ulogs/' . $filename)->spurt($testdata);
+    save_tmp_file($filename, $testdata);
+    assert_script_run('curl -O ' . autoinst_url . "/files/" . $filename);
+    my $sha1sum = sha1_sum(trim($testdata));    # cause script_output() trim the data
+    record_info("FILE", "SHA1: $sha1sum\nSIZE: $size");
+    $testdata = undef;                          # free
+
+    for (my $i = 0; $i < $repeat; $i++) {
+        my $output = undef;
+        eval {
+            script_output("echo 'RUN: $i'", quiet => 1, timeout => 10);
```

Comment:
why you need script_output here if you not collecting it ?

Reply:
True.

Reply:
so just replace this line with `script_run` and I will merge